### ISSUE TITLE
Integrity checking

### ISSUE THREAD

**Issue by ValtsS** [zbackup/zbackup#149]
Would it be possible to have a feature that would go over all files and verify their hashes, check that all files are present. As doing test-restores of all the files is not very feasible.

**Comment by davidbartonau** [comment]
@ValtsS I have started doing the latter, and it's time consuming but is feasible.  I have 100,000s of backups and I do a random sampling of them to check they restore.  zbackup won't restore them if there is a problem.

The only issue with this is that I can't check encrypted repositories on a server where there is no encryption key.

**Comment by davidbartonau** [comment]
If you would like the script let me know :-)

**Comment by ValtsS** [comment]
@davidbartonau I made a patch https://github.com/ValtsS/zbackup/commit/9383a664444ec1e75f301bf039a9885aac08b0dc to perform  loading of the indices/chunks. Tbh I am not certain how good that is of a check. But better than nothing.

**Comment by davidbartonau** [comment]
@ValtsS sounds good, I'd be interested in seeing it in action.  Have you done a pull request?

**Comment by ValtsS** [comment]
@davidbartonau no, because I am not certain if this is beneficial at all. I sadly don't have enough free time to have deep enough understanding of zbackup internals.

**Comment by davidbartonau** [comment]
@ValtsS I thought the interactive password was useful, although for me using an environment variable would be more useful as it is scriptable.

There are a lot of forks at the moment, it would be good to tie them all together.
